src/cryptonote_protocol/levin_notify.cpp

```diff
@@ -764,6 +770,34 @@ namespace levin
     );
   }
 
+  void notify::on_connection_new(const boost::uuids::uuid &id, bool is_income)
+  {
+    if (!zone_)
+      return;
+
+    const auto zone = zone_;
+    zone_->strand.dispatch([zone, id, is_income]{
```

Comment:
This copies `zone` twice, how about an explicit struct to prevent that ref-count destroying change? Its not really much code.